### PR TITLE
Updated sample

### PR DESCRIPTION
**Pull request by adrian-herscu**

### CODE REVIEW

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java [hunk 1]
```diff
@@ -1,58 +1,77 @@
...
-                    .phone(Phone.builder()
-                        .countryCode(1)
-                        .number(1)
-                        .build())
-                    .build())
-                .build(),
-        };
+        return Instancio.stream(createValidRequestModel())
+                .limit(100)
+                .toArray(CreatePersonRequest[]::new);
```
Comment (adrian-herscu) [inline src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:47]: There is no guarantee here that `Phone#countryCode` will be generated with value of 1, for example.

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java [hunk 2]
```diff
@@ -1,58 +1,77 @@
...
-                        .build())
-                    .build())
-                .build(),
-        };
+        // either empty or a string of 21+ characters in length
+        final StringSpec emptyStringOrLengthOver21 = Gen.string().allowEmpty().minLength(21);
+        final Generator<?> nullGenerator = random -> null;
+
+        // create an otherwise valid object, but with a single invalid field
+        final List<Tuple<TargetSelector, GeneratorSpec<?>>> negativeInputs = Arrays.asList(
```
Comment (adrian-herscu) [inline src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:57]: This means that if an annotation has changed then it has to be changed here too.
Imagine testing a REST application, implemented in other language. Yes, they may not have validation by annotations at all, and we cannot "reuse" their code at all since it is in other language.
So these annotatation are just for us to generate valid and invalid data.

The ability to generate a value based on field annotations was also requested here:
https://github.com/orgs/instancio/discussions/467#discussioncomment-5165705

**Review by adrian-herscu — COMMENTED** [review]
It still doesn't check limits and it requires manually updating the generator rules whenever annotations change.

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java [hunk 3]
```diff
@@ -1,58 +1,77 @@
...
+            Tuple.of(field(Address::getLine1), Gen.string().nullable().allowEmpty().minLength(21)), // line1 is not nullable
+            Tuple.of(field(Address::getLine2), emptyStringOrLengthOver21),
+            Tuple.of(field(Address::getCity), emptyStringOrLengthOver21),
+            Tuple.of(field(Address::getCountry), emptyStringOrLengthOver21),
+            Tuple.of(field(Address::getZipCode), emptyStringOrLengthOver21),
+            // random number outside the range [1-999]
+            Tuple.of(field(Phone::getCountryCode), Gen.ints().range(Integer.MIN_VALUE, 0).range(1000, Integer.MAX_VALUE)),
```
Comment (adrian-herscu) [inline src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:64]: This also means that whenever the model changes, e.g. a new class or field add/delete/chaged, these rules have to updated here, all in one place.

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java (hunk 1 above)
Comment (armandino) [reply src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:47]: That's right. It will generate a valid number, which is in the [1..999] range. The higher the sample size, the more permutations will be tested. But to test boundary conditions only, then the value would need to be specified manually (or maybe via some custom logic; I'm not sure how that might look like).

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java (hunk 2 above)
Comment (armandino) [reply src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:57]: That's true. If an annotation changes, then the invalid input to test that constraint would also change (especially if you're interested in boundary values, that is, the invalid value just outside the valid range).

The linked discussion is for generating valid data only. There's no out-of-the-box support for saying "given these constraints, generate an invalid object". That's somewhat outside the scope of the library, but could probably be implemented as an extension.

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java (hunk 3 above)
Comment (armandino) [reply src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:64]: That's right.

File: src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java (hunk 2 above)
Comment (adrian-herscu) [reply src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:57]: > The linked discussion is for generating valid data only. There's no out-of-the-box support for saying "given these constraints, generate an invalid object". That's somewhat outside the scope of the library, but could probably be implemented as an extension.

Ok... so how should I approach such "extension".
According to your comment
https://github.com/orgs/instancio/discussions/785#discussioncomment-7409535
The "generator provider SPI" might not be the best option. Then, what else?

Comment (armandino) [reply src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java:57]: I'm not really sure off the top of my head. It'll require some experimentation to see what works. Are you open to forking the project to implement this feature?
